src/channels/minesweeper/index.tsx

```diff
@@ -0,0 +1,238 @@
+
+function gridReducer(state: GridState, action: GridAction) {
+	switch (action.type) {
+		case actions.RESET: {
+			return generateInitialGridState();
+		}
+		case actions.FLAG_TILE: {
+			if (state.mines.length > 0) {
+				const mineIndexStr = randomFromArray(state.mines);
+				const [rowIndex, tileIndex] = splitTileIndex(mineIndexStr);
+
+				const newGrid = state.grid.map((row, ri) => {
+					if (ri !== rowIndex) return row;
+					return row.map<TileData>((tile, ti) =>
+						ti === tileIndex
+							? {
+									...tile,
+									tileType: TILE_MAP.FLAGGED,
+							  }
+							: tile,
+					);
+				});
+
+				const newMines = state.mines.filter((mineIndex) => mineIndex !== mineIndexStr);
+
+				return { ...state, grid: newGrid, mines: newMines };
+			}
+			return state;
+		}
+		case actions.REVEAL_TILES: {
+			if (state.nonMines.length > 0) {
+				const tileId = randomFromArray(state.nonMines);
+				const tiles = createTileCluster(state.grid, tileId, action.revealChance);
+
+				const grid = state.grid.map((row) => {
+					return row.map<TileData>((tile) => {
+						const mineCount = getMineCount(state.grid, tile.id);
+						return tiles.includes(tile?.id)
+							? {
+									...tile,
+									tileType: mineNumberTiles[mineCount],
+							  }
+							: tile;
+					});
+				});
+
+				const nonMines = state.nonMines.filter((id) => !tiles.includes(id));
+
+				return { ...state, grid, nonMines };
+			}
+			return state;
+		}
+		default: {
+			return state;
+		}
+	}
+}
+
+export function Minesweeper(props: ChannelProps) {
+	const [total] = useReplicant<Total | null>('total', null);
+
+	const [gridState, dispatch] = useReducer(gridReducer, undefined, generateInitialGridState);
+
+	const [face, setFace] = useState<Face>('smile');
+	const faceChangeTImeout = useRef<NodeJS.Timeout | undefined>(undefined);
+
+	useListenFor('donation', (donation: FormattedDonation) => {
+		/**
+		 * The magic numbers in here are sort of arbitrary placeholders since
+		 * I don't have much context for realistic donation behavior
+		 */
+		clearTimeout(faceChangeTImeout.current);
+		setFace(donation.rawAmount < 50 ? 'open_mouth' : 'sunglasses');
+		faceChangeTImeout.current = setTimeout(() => setFace('smile'), 2000);
+
+		if (donation.rawAmount < 50 && gridState.mines.length > 0) {
+			dispatch({ type: actions.FLAG_TILE });
+		} else {
+			// TODO: figure out which reveal chances to set for certain donation ranges
+			dispatch({ type: actions.REVEAL_TILES, revealChance: 0.5 });
+		}
+	});
+
+	useEffect(() => {
+		if (gridState.nonMines.length === 0) {
+			// reset the grid when we run out of hidden tiles to reveal
+			dispatch({ type: actions.RESET });
+		}
+	}, [gridState.nonMines]);
+
+	return (
+		<Container>
+			<Global
+				styles={css`
+					* {
+						box-sizing: border-box;
+					}
+				`}
+			/>
+			<Wrapper>
+				<Header>
+					<LCDContainer
+						css={css`
+							justify-content: flex-start;
+						`}>
+						<LCDText>SGDQ 2023</LCDText>
```

Comment:
Should instead use the ``shortname`` field of the ``currentEvent`` replicant, so that this doesn't need to be updated per event.

Reply:
Good catch, will do.